charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -2,7 +2,8 @@
 {{- if not .Values.client.grpc }}{{ fail "client.grpc must be true for api gateway" }}{{ end }}
```

Comment:
this will be repeated in all below helm files below, with `server-statefulset.yaml` also having a true logic change.

Reply:
I don't know if you need the check and bats tests for every file. 

If the settings are incorrect, helm will catch it right away in `server-statefulset.yaml` which makes `connect-inject-deployment.yaml`, etc redundant. 

Maybe keep it in the gateways as those can be deployed separately?

Reply:
i spoke about this repetition thing several months ago.  its hard to get around it because of the conditional installation of each file.  For example, if a kubernetes cluster is installed with only clients/components and server.enabled is false, then it will not pick up these settings.

Reply:
Sounds good!